Yamllint rules are clashing with ansible-lint's yamllint rules

## Describe the bug

Currently, when running super-linter without any config overrides, Ansible's yamllint rules are expecting one thing, while yamllint's yamllint rules are expecting another when it comes to spaces inside braces and brackets.

## Expected behavior

When running super-linter with the default settings (i.e. no config overrides), I expect super-linter's default rules for `yaml` and `ansible` to not clash.

## Steps to Reproduce

1. Create a file `./ansible/main.yml` with contents:

```yaml
---
- name: Main
  hosts: all

  tasks: []
```

2. Run the linter

```shell
docker run \
  -e RUN_LOCAL=true \
  -e YAML_ERROR_ON_WARNING=true \
  -e VALIDATE_ANSIBLE=TRUE \
  -e VALIDATE_YAML=true
  -v $(pwd):/tmp/lint \
  ghcr.io/github/super-linter:v4.9.2
```

`yamllint` has failed

<details>
  <summary>Click for the logs</summary>

```
2022-04-21 19:56:57 [ERROR]   Found errors in [yamllint] linter!
2022-04-21 19:56:57 [ERROR]   Error code: 2. Command output:
------
/tmp/lint/ansible/main.yml:5:11: [warning] too few spaces inside empty brackets (brackets)
------
2022-04-21 19:56:57 [INFO]   ----------------------------------------------
2022-04-21 19:56:57 [INFO]   ----------------------------------------------
2022-04-21 19:56:57 [INFO]   The script has completed
2022-04-21 19:56:57 [INFO]   ----------------------------------------------
2022-04-21 19:56:57 [INFO]   ----------------------------------------------
2022-04-21 19:56:58 [ERROR]   ERRORS FOUND in YAML:[1]
2022-04-21 19:56:58 [FATAL]   Exiting with errors found!
```

</details>

3. Fix the issue in the file

```yaml
---
- name: Main
  hosts: all

  tasks: [ ]
```

4. Run the linter again

```shell
docker run \
  -e RUN_LOCAL=true \
  -e YAML_ERROR_ON_WARNING=true \
  -e VALIDATE_ANSIBLE=TRUE \
  -e VALIDATE_YAML=true
  -v $(pwd):/tmp/lint \
  ghcr.io/github/super-linter:v4.9.2
```

Now `ansible` has failed (through its integrated yamllint), while the `yamllint` linter does not fail

<details>
  <summary>Click for the logs</summary>

```
WARNING  Listing 1 violation(s) that are fatal
ansible/main.yml:5: yaml (brackets)
------
WARNING  Listing 1 violation(s) that are fatal
ansible/main.yml:5: yaml (brackets)
------
2022-04-21 19:57:28 [INFO]   
2022-04-21 19:57:28 [INFO]   ----------------------------------------------
2022-04-21 19:57:28 [INFO]   ----------------------------------------------
2022-04-21 19:57:28 [INFO]   Linting [YAML] files...
2022-04-21 19:57:28 [INFO]   ----------------------------------------------
2022-04-21 19:57:28 [INFO]   ----------------------------------------------
2022-04-21 19:57:28 [INFO]   ---------------------------
2022-04-21 19:57:28 [INFO]   File:[/tmp/lint/ansible/main.yml]
2022-04-21 19:57:28 [INFO]    - File:[main.yml] was linted with [yamllint] successfully
2022-04-21 19:57:28 [INFO]   ----------------------------------------------
2022-04-21 19:57:28 [INFO]   ----------------------------------------------
2022-04-21 19:57:28 [INFO]   The script has completed
2022-04-21 19:57:28 [INFO]   ----------------------------------------------
2022-04-21 19:57:28 [INFO]   ----------------------------------------------
2022-04-21 19:57:28 [ERROR]   ERRORS FOUND in ANSIBLE:[1]
2022-04-21 19:57:28 [FATAL]   Exiting with errors found!
```

</details>

## Possible solutions

There are a few solutions I can think of:

1. [Add the `yaml` tag to skip_list](https://ansible-lint.readthedocs.io/en/latest/configuring/#configuration-file) for ansible as part of the default `.ansible-lint.yml`. It can be argued that ansible's yamllint is not needed in a linter aggregate that already contains a dedicated yaml linter.
2. Synchronise the relevant configuration properties by updating the default `.yaml-lint.yml`. At the minimum the `braces.min-spaces-inside-empty` and `brackets.min-spaces-inside-empty` would need to be set to `-1` to match the defaults of ansible's yamllint. Ideally all the `*-spaces-*` configuration properties are set to the ansible's yamllint defaults inside the default `.yaml-lint.yml`.
3. Extend the default yamllint config with `extends: default` and remove all the `*-spaces-*` properties from `.yaml-lint.yml`. This is the most foolproof option in terms of automatically receiving new default configuration properties but it's also the most destructive as it could pull in lots of unexpected properties.

I am happy to create a PR for any of the above.

## Additional context

This is one of the few discrepancies that I noticed between `ansible`'s yamllint config super-linter's `yamllint` config and the default yamllint config. There are quite a few more, I have included all configs for comparison:

<details>
  <summary>Click for ansible's yamllint config</summary>
  
  ```yaml
  ---
  rules:
    braces:
      forbid: false
      level: error
      max-spaces-inside: 0
      max-spaces-inside-empty: -1
      min-spaces-inside: 0
      min-spaces-inside-empty: -1
    brackets:
      forbid: false
      level: error
      max-spaces-inside: 0
      max-spaces-inside-empty: -1
      min-spaces-inside: 0
      min-spaces-inside-empty: -1
    colons:
      level: error
      max-spaces-after: 1
      max-spaces-before: 0
    commas:
      level: error
      max-spaces-after: 1
      max-spaces-before: 0
      min-spaces-after: 1
    comments:
      ignore-shebangs: true
      level: warning
      min-spaces-from-content: 1
      require-starting-space: true
    comments-indentation: false
    document-end: false
    document-start: false
    empty-lines:
      level: error
      max: 2
      max-end: 0
      max-start: 0
    empty-values: false
    hyphens:
      level: error
      max-spaces-after: 1
    indentation:
      check-multi-line-strings: false
      indent-sequences: true
      level: error
      spaces: consistent
    key-duplicates:
      level: error
    key-ordering: false
    line-length:
      allow-non-breakable-inline-mappings: false
      allow-non-breakable-words: true
      level: error
      max: 160
    new-line-at-end-of-file:
      level: error
    new-lines:
      level: error
      type: unix
    octal-values: false
    quoted-strings: false
    trailing-spaces:
      level: error
    truthy:
      allowed-values:
      - 'true'
      - 'false'
      check-keys: true
      level: warning
  ```
</details>

<details>
  <summary>Click for super-linter's yamllint config</summary>
  
  ```yaml
  ---
  rules:
    braces:
      level: warning
      max-spaces-inside: 0
      max-spaces-inside-empty: 5
      min-spaces-inside: 0
      min-spaces-inside-empty: 1
    brackets:
      level: warning
      max-spaces-inside: 0
      max-spaces-inside-empty: 5
      min-spaces-inside: 0
      min-spaces-inside-empty: 1
    colons:
      level: warning
      max-spaces-after: 1
      max-spaces-before: 0
    commas:
      level: warning
      max-spaces-after: 1
      max-spaces-before: 0
      min-spaces-after: 1
    comments: disable
    comments-indentation: disable
    document-end: disable
    document-start:
      level: warning
      present: true
    empty-lines:
      level: warning
      max: 2
      max-end: 0
      max-start: 0
    hyphens:
      level: warning
      max-spaces-after: 1
    indentation:
      check-multi-line-strings: false
      indent-sequences: true
      level: warning
      spaces: consistent
    key-duplicates: enable
    line-length:
      allow-non-breakable-inline-mappings: true
      allow-non-breakable-words: true
      level: warning
      max: 80
    new-line-at-end-of-file: disable
    new-lines:
      type: unix
    trailing-spaces: disable
  ```
</details>


<details>
  <summary>Click for default yamllint config</summary>
  
  ```yaml
  ---
  rules:
    braces:
      forbid: false
      level: error
      max-spaces-inside: 0
      max-spaces-inside-empty: -1
      min-spaces-inside: 0
      min-spaces-inside-empty: -1
    brackets:
      forbid: false
      level: error
      max-spaces-inside: 0
      max-spaces-inside-empty: -1
      min-spaces-inside: 0
      min-spaces-inside-empty: -1
    colons:
      level: error
      max-spaces-after: 1
      max-spaces-before: 0
    commas:
      level: error
      max-spaces-after: 1
      max-spaces-before: 0
      min-spaces-after: 1
    comments:
      ignore-shebangs: true
      level: warning
      min-spaces-from-content: 2
      require-starting-space: true
    comments-indentation:
      level: warning
    document-end: false
    document-start:
      level: warning
      present: true
    empty-lines:
      level: error
      max: 2
      max-end: 0
      max-start: 0
    empty-values: false
    hyphens:
      level: error
      max-spaces-after: 1
    indentation:
      check-multi-line-strings: false
      indent-sequences: true
      level: error
      spaces: consistent
    key-duplicates:
      level: error
    key-ordering: false
    line-length:
      allow-non-breakable-inline-mappings: false
      allow-non-breakable-words: true
      level: error
      max: 80
    new-line-at-end-of-file:
      level: error
    new-lines:
      level: error
      type: unix
    octal-values: false
    quoted-strings: false
    trailing-spaces:
      level: error
    truthy:
      allowed-values:
      - 'true'
      - 'false'
      check-keys: true
      level: warning
  ```
</details>

--- a/TEMPLATES/.ansible-lint.yml
+++ b/TEMPLATES/.ansible-lint.yml
@@ -36,6 +36,7 @@ skip_list:
   - 'command-instead-of-shell'  # Allow use of shell when you want
   - 'no-handler'  # Allow step to run like handler
   - 'unnamed-task' # Allow tasks without a name
+  - 'yaml' # Disable YAML linting since it's done by yamllint
 
 ##################
 # Tags to follow #

